lib/src/dev/mt_af_xdp.c

```diff
@@ -963,14 +998,17 @@ struct mt_rx_xdp_entry* mt_rx_xdp_get(struct mtl_main_impl* impl, enum mtl_port
 }
 
 int mt_rx_xdp_put(struct mt_rx_xdp_entry* entry) {
+  struct mtl_main_impl* impl = entry->parent;
   enum mtl_port port = entry->port;
   struct mt_rxq_flow* flow = &entry->flow;
   uint8_t* ip = flow->dip_addr;
   struct mt_xdp_queue* xq = entry->xq;
 
   if (entry->flow_rsp) {
-    mt_rx_flow_free(entry->parent, port, entry->flow_rsp);
+    mt_rx_flow_free(impl, port, entry->flow_rsp);
     entry->flow_rsp = NULL;
+    if (!mt_is_privileged(impl))
+      xdp_socket_update_dp(mt_kernel_if_name(impl, port), flow->dst_port, false);
```

Comment:
nit: how to update the UDP filter to BPF in privileged mode?

Reply:
Good catch, will figure out when the new daemon server is designed. Right now with privileged mode we assume that et is not used, so the default xdp program does not support filter.

Reply:
Checked again, if we load custom xdp prog in et then start with root, the default prog will not be attached. Will change the logic here to detect if the server connected.